nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/storage/AbstractGCSProcessor.java

```diff
@@ -141,6 +142,10 @@ protected StorageOptions getServiceOptions(ProcessContext context, GoogleCredent
             storageOptionsBuilder.setProjectId(projectId);
         }
 
+        if (storageApiHost != null && !storageApiHost.isEmpty()) {
+            storageOptionsBuilder.setHost(storageApiHost);
```

Comment:
According to the [Private Service Connect for Google APIs](https://codelabs.developers.google.com/cloudnet-psc#12) documentation, the HTTP `Host` header should also be overridden to send `www.googleapis.com`.